docs/content/doc/advanced/config-cheat-sheet.en-us.md

```diff
@@ -568,7 +568,21 @@ Certain queues have defaults that override the defaults set in `[queue]` (this o
 - `IMPORT_LOCAL_PATHS`: **false**: Set to `false` to prevent all users (including admin) from importing local path on server.
 - `INTERNAL_TOKEN`: **\<random at every install if no uri set\>**: Secret used to validate communication within Gitea binary.
 - `INTERNAL_TOKEN_URI`: **<empty>**: Instead of defining INTERNAL_TOKEN in the configuration, this configuration option can be used to give Gitea a path to a file that contains the internal token (example value: `file:/etc/gitea/internal_token`)
-- `PASSWORD_HASH_ALGO`: **pbkdf2**: The hash algorithm to use \[argon2, pbkdf2, scrypt, bcrypt\], argon2 will spend more memory than others.
+- `PASSWORD_HASH_ALGO`: **pbkdf2**: The hash algorithm to use \[argon2, pbkdf2, pbkdf2_v1, scrypt, bcrypt\], argon2 and scrypt will spend significant amounts of memory.
+  - Note: The default parameters for `pbkdf2` hashing have changed - the previous settings are available as `pbkdf2_v1` but are not recommended.
```

Comment:
>  The default parameters for `pbkdf2` hashing have changed

If  the default parameters have been changed, does the new pbkdf2 work with old passwords?

I can see this in my database for 123456:

```
password:  95b9469d174e39509b30652eeb58c9e03ce240d3ce201ad4f5e5b0c93f8226c90776660f0e3a4fb7e21597c5e55c0be9c39e
algo: pbkdf2
salt: 93d7d5c0d0b0df2d40ad664dbf4c4808
```

Reply:
It transparently just rehashes and works. It would be extremely breaking otherwise and would be clearly inappropriate.

Reply:
OK, I see the ambiguity part. 

For config: pbkdf2 is default to pbkdf2_v2

But for code (NewPBKDF2Hasher), pbkdf2 is default to pbkdf2_v1 .....